src/components/IconButton.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { PropsWithChildren, ReactElement, useContext } from "react";
+import { Button, ConfigProvider } from "antd";
+import { ThemeContext } from "./AppStyle";
+import styled, { css } from "styled-components";
+
+type IconButtonProps = {
+  onClick?: () => void;
+  disabled?: boolean;
+  type?: "outlined" | "primary";
+};
+
+// Button styling varies based on the type (outlined vs. primary)
+const StyledButton = styled(Button)<{ $type: IconButtonProps["type"] }>`
+  height: var(--button-height-small);
+  width: var(--button-height-small);
+  min-width: var(--button-height-small);
+  border-radius: 4px;
+  padding: 0;
+
+  ${(props) => {
+    switch (props.$type) {
+      case "outlined":
+        return css`
+          border: 1px solid var(--color-button);
+          background-color: transparent;
+          color: var(--color-button);
+          fill: var(--color-button);
+
+          &:disabled {
+            border: 1px solid var(--color-borders);
+            background-color: var(--color-button-disabled);
+            color: var(--color-text-disabled);
+            fill: var(--color-text-disabled);
+          }
+        `;
+      case "primary":
+      default:
+        return css`
+          fill: var(--color-text-button);
+
+          &:disabled {
+            fill: var(--color-text-disabled);
+          }
+        `;
+    }
+  }}
+
+  &:not(:disabled):hover {
+    border-color: var(--color-button-hover);
+    color: var(--color-text-button);
+    fill: var(--color-text-button);
+  }
```

Comment:
This uses styled-component's syntax. `&` selects this current element.

Reply:
Can you explain the choice of why to use styled-components?  I understand there are pros and cons for maintainability of code.

Reply:
I primarily added it for the changes I made in `AppStyle.tsx`, namely that I can have variables be shared across CSS and Javascript for styling and theming. I think there are other ways to do it too (`element.style.setProperty("--some-var", value)`) but this approach is cleaner.

I switched a couple smaller components to also use styled-components where the CSS was relatively trivial. (Though, admittedly, it's not super consistent!) One of the nice things about styled-components is that you're not relying on having to match classnames between documents, so it's easier to tell if CSS is actually being used (everything lives in one document).